### PR TITLE
AIP-84: Migrating DELETE queued asset events for DAG to fastAPI

### PR DESCRIPTION
**Pull request by amoghrajesh**
related: https://github.com/apache/airflow/issues/42370
Migrating delete queued asset events for DAG to fastAPI 

Dependent on https://github.com/apache/airflow/pull/44124

Same setup as https://github.com/apache/airflow/pull/44124

Responses:
1. Legacy
image

2. FastAPI
image

With time filtering
1. Legacy
image

2. FastAPI
image

Time filtering but no queued event found
1. Legacy
image

2. FastAPI
image

---
**^ Add meaningful description above**
Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/contributing-docs/05_pull_requests.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

File: tests/api_fastapi/core_api/routes/public/test_assets.py [hunk 1]
```diff
@@ -464,16 +479,77 @@ def test_should_respond_404(self, test_client):
...
+            f"/public/dags/{dag_id}/assets/queuedEvent",
+        )
+
+        assert response.status_code == 204
+
+    def test_should_respond_404(self, test_client):
+        dag_id = "not_exists"
+
+        response = test_client.delete(
+            f"/public/dags/{dag_id}/assets/queuedEvent",
```
Comment (rawwar) [inline tests/api_fastapi/core_api/routes/public/test_assets.py:549]: Shouldn't we test this with an existing DAG that has no queued events?

File: tests/api_fastapi/core_api/routes/public/test_assets.py [hunk 2]
```diff
@@ -523,6 +523,32 @@ def test_should_respond_404(self, test_client):
...
+        dag_id = dag.dag_id
+        self.create_assets(session=session, num=1)
+        asset_id = 1
+        self._create_asset_dag_run_queues(dag_id, asset_id, session)
+
+        response = test_client.delete(
+            f"/public/dags/{dag_id}/assets/queuedEvent",
+        )
+
+        assert response.status_code == 204
```
Comment (Lee-W) [inline tests/api_fastapi/core_api/routes/public/test_assets.py:541]: I think we should also check the event is deleted

**Review by Lee-W — CHANGES_REQUESTED** [review]

File: tests/api_fastapi/core_api/routes/public/test_assets.py (hunk 2 above)
Comment (amoghrajesh) [reply tests/api_fastapi/core_api/routes/public/test_assets.py:541]: Sure, that would be a nice validation to have.

File: tests/api_fastapi/core_api/routes/public/test_assets.py (hunk 1 above)
Comment (amoghrajesh) [reply tests/api_fastapi/core_api/routes/public/test_assets.py:549]: The test was migrated from legacy, both the cases are valid:
1. invalid dag 
2. valid dag but no queued events.
I will add test case for the second one

Comment (amoghrajesh) [reply tests/api_fastapi/core_api/routes/public/test_assets.py:549]: I pushed a new test case that should cover this

File: tests/api_fastapi/core_api/routes/public/test_assets.py (hunk 2 above)
Comment (amoghrajesh) [reply tests/api_fastapi/core_api/routes/public/test_assets.py:541]: Pushed a fix, can you check?

**Comment by amoghrajesh** [comment]
@Lee-W @rawwar I pushed a fix for the reviews, can you check again?

**Review by Lee-W — APPROVED** [review]

**Review by pierrejeambrun — APPROVED** [review]
👍

**Pull request merged**